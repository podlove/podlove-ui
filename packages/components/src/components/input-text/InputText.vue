<template>
  <input
    class="input-text box-border inline-block resize-none p-1 border border-solid rounded-sm h-8"
    :class="{ 'w-full block': block }"
    :disabled="disabled"
    :value="value"
    :style="style"
  />
</template>

<script>
import { color, background } from '../../defaults'

export default {
  props: {
    disabled: {
      type: Boolean,
      default: false
    },

    value: {
      type: String,
      default: ''
    },

    color: {
      type: String,
      default: background
    },

    borderColor: {
      type: String,
      default: background
    },

    background: {
      type: String,
      default: color
    },

    block: {
      type: Boolean,
      default: false
    }
  },
  computed: {
    style() {
      return {
        color: this.color,
        background: this.background,
        'border-color': this.borderColor
      }
    }
  }
}
</script>

<style lang="scss" scoped></style>
