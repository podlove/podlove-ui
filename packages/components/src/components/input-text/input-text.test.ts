import { describe, test, expect } from 'vitest';
import { mount } from '@vue/test-utils';

import InputText from './InputText.vue';

describe('InputText', () => {
  describe('component', () => {
    test('should be a Vue instance', async () => {
      const wrapper = await mount(InputText);

      expect(wrapper.vm).toBeTruthy();
    });
  });

  describe('props', () => {
    test('should render without props', async () => {
      const wrapper = await mount(InputText);
      expect(wrapper.element).toMatchSnapshot();
    });

    const props = [
      {
        prop: 'disabled',
        value: true
      },
      {
        prop: 'value',
        value: 'input-value'
      },
      {
        prop: 'color',
        value: '#fff'
      },
      {
        prop: 'borderColor',
        value: '#fff'
      },
      {
        prop: 'background',
        value: '#fff'
      },
      {
        prop: 'block',
        value: true
      }
    ];

    props.forEach(({ prop, value }) => {
      test(`should render type '${prop}'`, async () => {
        const wrapper = await mount(InputText, {
          props: {
            [prop]: value
          }
        });

        expect(wrapper.element).toMatchSnapshot();
      });
    });
  });
});
