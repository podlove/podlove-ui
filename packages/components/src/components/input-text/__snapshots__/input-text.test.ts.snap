// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`InputText > props > should render type 'background' 1`] = `
<input
  background="#fff"
  class="podlove-component-input-text box-border inline-block resize-none p-1 border border-solid rounded-sm h-8"
  data-v-41de188a=""
/>
`;

exports[`InputText > props > should render type 'block' 1`] = `
<input
  block="true"
  class="podlove-component-input-text box-border inline-block resize-none p-1 border border-solid rounded-sm h-8"
  data-v-41de188a=""
/>
`;

exports[`InputText > props > should render type 'borderColor' 1`] = `
<input
  bordercolor="#fff"
  class="podlove-component-input-text box-border inline-block resize-none p-1 border border-solid rounded-sm h-8"
  data-v-41de188a=""
/>
`;

exports[`InputText > props > should render type 'color' 1`] = `
<input
  class="podlove-component-input-text box-border inline-block resize-none p-1 border border-solid rounded-sm h-8"
  color="#fff"
  data-v-41de188a=""
/>
`;

exports[`InputText > props > should render type 'disabled' 1`] = `
<input
  class="podlove-component-input-text box-border inline-block resize-none p-1 border border-solid rounded-sm h-8"
  data-v-41de188a=""
  disabled=""
/>
`;

exports[`InputText > props > should render type 'value' 1`] = `
<input
  class="podlove-component-input-text box-border inline-block resize-none p-1 border border-solid rounded-sm h-8"
  data-v-41de188a=""
/>
`;

exports[`InputText > props > should render without props 1`] = `
<input
  class="podlove-component-input-text box-border inline-block resize-none p-1 border border-solid rounded-sm h-8"
  data-v-41de188a=""
/>
`;
